number-of-islands.java

```diff
@@ -0,0 +1,35 @@
+number of islands solution DFS solution:
+
+//this is a DFS solution, if we find a island mark it as 0 and check its neighbouring land, keep marking all as 0 until there is no land left.
```

Comment:
Very good. Appreciate you for trying out DFS approach, hope you also know how to solve this using BFS. Pls mention the Time and Space complexities